freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UsersState.java

```diff
@@ -15,10 +17,14 @@
 
     private Multimap<String, String> groupMembership;
 
-    public UsersState(Set<FmsGroup> groups, Set<FmsUser> users, Multimap<String, String> groupMembership) {
+    private Map<String, WorkloadCredential> usersWorkloadCredentialMap;
```

Comment:
I think we should split this off from the UsersState somehow because it isn't something we would expect back from the ipa server. Maybe subclass UsersState to a UmsState class? There's other information from the UMS that we would want to hold on to, like the user crn --> workloadUsername mapping. We'll need that for fixing the single user sync. Creating a UmsState class would give us a place to hold any extra UMS information we need.

Reply:
+1 
We need to re-visit and have more interfaces rather than concrete implementations also child-parent relationships. Can be a separate PR.